### ISSUE TITLE
Missing ForeignObjectRel support for wagtail.wagtailsearch.index.RelatedFields

### ISSUE THREAD

**Issue by lucasmoeskops** [wagtail/wagtail#2493]
Hello,

I'm trying to make an index setup similar to the Books / Authors example from the Wagtail docs: http://docs.wagtail.io/en/v1.4.3/topics/search/indexing.html.
The Author model with a ForeignObjectRel to Books doesn't work for me though when indexing. The related content doesn't get indexed. 

Upon inspecting, it seems to me that there is missing support for ForeignObjectRel[ations] in the method wagtail.wagtailsearch.index.RelatedFields.get_value.
When comparing to the wagtail.wagtailsearch.index.RelatedFields.select_on_queryset method, the latter does have support for both RelatedField and ForeignObjectRel. Also, when comparing to the elasticsearch backend, which I am using, the backend seems to be able to deal with it as well. If I modify the get_value method to handle ForeignObjectRel as well, the indexing problem is solved:

Old:

```
def get_value(self, obj):
        field = self.get_field(obj.__class__)

        if isinstance(field, RelatedField):
            return getattr(obj, self.field_name)
```

New:

```
def get_value(self, obj):
        field = self.get_field(obj.__class__)

        if isinstance(field, RelatedField):
            return getattr(obj, self.field_name)

        if isinstance(field, ForeignObjectRel):
            return getattr(obj, self.field_name)
```

In the new version, I added a simple support for ForeignObjectRel. Upon indexing using the elasticsearch backend, the related content is now correctly indexed.

Regards,
Lucas Moeskops

**Comment by gasman** [comment]
Hi @lucasmoeskops - thanks for the report! Please can you create a unit test for this, to confirm that the fix works (and to ensure that we don't break it again in future)?

I guess this will involve adding a new test to wagtail/wagtailsearch/tests/test_elasticsearch_backend.py, similar to https://github.com/torchbox/wagtail/blob/9e7c6204dbfdd8a6b94216570aa17249b1cc1968/wagtail/wagtailsearch/tests/test_elasticsearch_backend.py#L781-L808 (or extending the existing test) - it might also require some additions to https://github.com/torchbox/wagtail/blob/master/wagtail/tests/search/models.py, if it doesn't already have the appropriate kind of relation.

**Comment by gasman** [comment]
PR: #2547

**Comment by BertrandBordage** [comment]
Fixed by 90a62523d1055af15f1da40b2047dd667d6e023c for elasticsearch & 1943d5a83c8a8a68345cf48bee9b739ad4001aad for PostgreSQL.
Still not implemented for the generic database backend, but I think we can put that aside and close for now.

Ideally, we should create a table comparing features of different backends to make it perfectly clear which feature is supported by which backend.
